scripts/configure-netrc-for-mapbox-sdk-installation.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash
+
+# Mapbox SDK Configuration Script for iOS and Android
+# ===================================================
+#
+# Purpose:
+# --------
+# This script configures the development environment to download Mapbox SDKs 
+# for both iOS and Android builds. We use Mapbox to display maps in the App. As Mapbox SDKs 
+# are closed-sourced, we need to authenticate with Mapbox during the download.
+#
+# Background:
+# -----------
+# Engineers are required to obtain a secret token from Mapbox and store it on 
+# their development machine. This allows tools like CocoaPods for iOS or Gradle for Android 
+# to access the Mapbox SDK during the build process.
+#
+# The `.netrc` file for iOS Configuration:
+# ----------------------------------------
+# The token for iOS is stored in the `.netrc` file located in the user's home directory.
+# This file is used in Unix-like systems to store credentials for remote machine access.
+#
+# The `gradle.properties` file for Android Configuration:
+# -------------------------------------------------------
+# The token for Android is stored in the `gradle.properties` file located in the .gradle directory 
+# in the user's home. This is accessed by the Android build system during the SDK download.
+#
+# How this script helps:
+# ----------------------
+# This script streamlines the process of adding the credential to both the `.netrc` and 
+# `gradle.properties` files. When executed, it prompts the user for the secret token and 
+# then saves it to the respective files along with other necessary information.\n
+
+NETRC_PATH="$HOME/.netrc"
+GRADLE_PROPERTIES_PATH="$HOME/.gradle/gradle.properties"
+
+# This function provides a user-friendly error message when the script encounters an error.
+# It informs the user about probable permission issues and suggests commands to resolve them.
+handleError() {
+    echo -e "\n"
+
+    echo -e "\033[1;31mError:\033[0m The script failed."
+    echo "The most probable reason is permissions."
+    echo -e "Please ensure you have read/write permissions for the following:\n"
+    
+    echo -e "1. \033[1m$NETRC_PATH\033[0m"
+    echo -e "2. \033[1m$GRADLE_PROPERTIES_PATH\033[0m"
+    echo -e "\nYou can grant permissions using the commands:"
+    echo -e "\033[1mchmod u+rw $NETRC_PATH\033[0m"
+    echo -e "\033[1mchmod u+rw $GRADLE_PROPERTIES_PATH\033[0m"
+
+    echo -e "\n"
+    exit 1
+}
+
+# Set a trap to call the handleError function when any of the commands fail
+trap handleError ERR
+
+# Prompt the user for the token
```

Comment:
I created another script `configure-mapbox-walkthrough.sh` that interactively instructs the user to create a Mapbox account, create a token, and paste created token to the command prompt.

Once the token is corrected. it invokes and pass token to `configure-mapbox.sh`, which actually sets the token in relevant files. Because you pass the token as a command argument to `configure-mapbox.sh`, you can use it in CI, too.